src/main/java/org/jabref/logic/importer/fetcher/WorldcatFetcher.java

```diff
@@ -0,0 +1,216 @@
+ * 
+ * @see https://www.oclc.org/developer/develop/web-services/worldcat-search-api/bibliographic-resource.en.html
+ */
+public class WorldcatFetcher implements EntryBasedFetcher {
+
+    private String WORLDCAT_OPEN_SEARCH_URL = "http://www.worldcat.org/webservices/catalog/search/opensearch?wskey=";
+    private String WORLDCAT_READ_URL = "http://www.worldcat.org/webservices/catalog/content/{OCLC-NUMBER}?recordSchema=info%3Asrw%2Fschema%2F1%2Fdc&wskey=";
+   
+
+    public WorldcatFetcher (String worldcatKey) {
+        WORLDCAT_OPEN_SEARCH_URL += worldcatKey;
+        WORLDCAT_READ_URL += worldcatKey;
+    }
+
+    @Override
+    public String getName () {
+        return "Worldcat Fetcher";
+    }
+
+    /**
+     * Create a open search query with specified title
+     * @param title the title to include in the query
+     * @return the earch query for the api
+     */
+    private String getOpenSearchURL (String title) throws MalformedURLException { 
+        String query = "&q=srw.ti+all+\"" + title + "\"";
+        URL url = new URL (WORLDCAT_OPEN_SEARCH_URL + query);
+        return url.toString ();
+    }
+
+    /**
+     * Make request to open search API of Worldcat, with specified title
+     * @param title the title of the search
+     * @return the body of the HTTP response
+     */
+    private String makeOpenSearchRequest (String title) throws FetcherException { 
+        try {
+            URLDownload urlDownload = new URLDownload (getOpenSearchURL (title));
+            URLDownload.bypassSSLVerification ();
+            String resp = urlDownload.asString ();
+
+            return resp;
+        } catch (MalformedURLException e) {
+            throw new FetcherException ("Bad url", e);
+        } catch (IOException e) {
+            throw new FetcherException ("Error with Open Search Request (Worldcat)", e);
+        } 
+    }
+
+     /**
+     * Get more information about a article through its OCLC id. Picks the first 
+     * element with this tag
+     * @param id the oclc id
+     * @return the XML element that contains all tags
+     */
+    private Element getSpecificInfoOnOCLC (String id) throws IOException {
+        URLDownload urlDownload = new URLDownload (WORLDCAT_READ_URL.replace ("{OCLC-NUMBER}", id));
+        URLDownload.bypassSSLVerification ();
+        String resp = urlDownload.asString ();	
+
+        Document mainDoc = parse (resp);
+        NodeList parentElemOfTags = mainDoc.getElementsByTagName ("oclcdcs");
+
+        return (Element) parentElemOfTags.item (0);
+    }
+    
+     /**
+     * Parse a string to an xml document
+     * @param s the string to be parsed
+     * @return XML document representing the content of s
+     * @throws IllegalArgumentException if s is badly formated or other exception occurs during parsing
+     */
+    private Document parse (String s) {
+        try { 
+            BufferedReader r = new BufferedReader (new StringReader(s));
```

Comment:
You never close the BufferedReader, this will create a ResourceLeak, wrap it in a try with resource statement
```
  try (BufferedReader br = new BufferedReader){
....
}
```